src/query/service/src/interpreters/common/grant.rs

```diff
@@ -28,6 +29,13 @@ pub async fn validate_grant_object_exists(
 
     match &object {
         GrantObject::Table(catalog_name, database_name, table_name) => {
+            if ProcedureFactory::instance()
```

Comment:
It would be better to add a  procedure privilege test in tests/suites/0_stateless/20+_others/20_0012_privilege_access.sh

Reply:
Else lgtm, 💯

Reply:
done